pkg/otlp/collector.go

```diff
@@ -206,6 +206,14 @@ func NewPipeline(cfg PipelineConfig, s serializer.MetricSerializer, logsAgentCha
 
 // Run the OTLP pipeline.
 func (p *Pipeline) Run(ctx context.Context) error {
+	defer func() {
+		// Recover any panic that happened during execution.
+		if r := recover(); r != nil {
+			err := fmt.Errorf("OTLP pipeline had a panic: %v", r)
+			pipelineError.Store(err)
+			log.Errorf(err.Error())
+		}
```

Comment:
Shouldn't we restart the pipeline after recover ?

Reply:
I don't think we should unless other Agent components do; this opens a can of worms (what strategy should we follow? should we let the user configure the strategy? What if the failure stems from a different component in the Agent?) and this is something that is typically handled by an external application such as Kubernetes. The Agent (and therefore its subcomponents) should have a way to report its status to the orchestrator, but not handle the restarts itself.

It's also independent from this change; we currently don't do restarts for any kinds of errors (including those that are not panics), I think if we start to do restarts we should also consider those.